src/lib/util/textfile.c

```diff
@@ -190,3 +191,86 @@ textfile_write(const char *filepath,
 
     return ret;
 }
+
+errno_t
+textfile_copy(const char *source,
+              const char *destdir,
+              const char *destname,
+              mode_t dir_mode,
+              mode_t file_mode)
```

Comment:
This is probably OK for backup, but you should keep in mind that the owner would be root in this case (or whoever is running authselect), so the restore needs to deal with the ownership.

Reply:
These files are owned usually by root anyway, but if you think it is needed, I can chown and chmod it.

Reply:
I think the burden will be on whoever is doing the restore, I'm just trying to say that if e.g. IPA is going to restore the files, we need to make sure they take care about things like ownership.

Reply:
I changed the new patches so the owner and permission is kept.